sdk/core/azure-core/src/main/java/com/azure/android/core/http/interceptor/UserAgentInterceptor.java

```diff
@@ -3,13 +3,19 @@
 
 package com.azure.android.core.http.interceptor;
 
+import android.content.Context;
```

Comment:
Whenever we need to reference a class in the `android` package from within some business logic (in this case, the user agent interceptor), I've found it extremely beneficial to create an interface to decouple them. If we don't decouple them, we're unable to unit-test the UserAgentInterceptor. So instead I might imagine:

```java
package com.azure.android.core.provider;
public interface AppMetadataProvider {
    public String getApplicationVersion();
    public String getTargetSdkVersion();
    ...
}

---

package com.azure.android.core.provider.impl;
public class AndroidAppMetadataProvider implements AppMetadataProvider {
    public AndroidAppMetadataProvider(Context context) {
        this.appInfo = context.getApplicationInfo();
        ...
    }

    public String getTargetSdkVersion() {
        return Integer.toString(this.appInfo.targetSdkVersion);
    }

    ...
}

---

package com.azure.android.core.http.interceptor;
public class UserAgentInterceptor implements Interceptor {
    public UserAgentInterceptor(AppMetadataProvider provider, String userAgent) {
        ...
    }
}
```

Using DI here to decouple the Android system APIs lets us easily write unit tests by defining and passing in a `TestAppMetadatProvider` for unit tests that can return stub values.

Reply:
To clarify, the example above is the simple case, where we only need to avoid calling Platform API methods. If the API methods also return an instance of a Platform Type, we would need to wrap that type as well.

Reply:
Thank you for the thorough explanation! Will add these changes later this week.